compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -40,6 +40,34 @@ pub trait IntoDiagnosticArg {
     fn into_diagnostic_arg(self) -> DiagnosticArgValue<'static>;
 }
 
+macro_rules! into_diagnostic_arg_using_tostring {
+    ($( $ty:ty ),+ $(,)?) => {
+        $(
+            impl IntoDiagnosticArg for $ty {
+                fn into_diagnostic_arg(self) -> DiagnosticArgValue<'static> {
+                    self.to_string().into_diagnostic_arg()
+                }
+            }
+        )+
+    }
+}
+
+into_diagnostic_arg_using_tostring!(
```

Comment:
I wonder if `into_diagnostic_arg_using_display` might be clearer here, since most (all?) of these implement `ToString` via the blanket impl of `Display`

Reply:
That makes sense, will change this.